hathor/execution_manager.py

```diff
@@ -0,0 +1,51 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import os
+from typing import NoReturn
+
+from structlog import get_logger
+
+from hathor.event import EventManager
+from hathor.transaction.storage import TransactionStorage
+
+logger = get_logger()
+
+
+class ExecutionManager:
```

Comment:
Maybe add a `register_on_crash(self, callback, *, priority)` that will keep record of all calls to be executed when the full node crashes. You might use the `priority` to order the calls.

Reply:
Notice that we have to handle all exceptions raised by these callbacks so the `exit()` method is called.

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/809/commits/2938c0855a3efcceece808b8116061e25489dce9

Reply:
Great! Thanks!